Dockerfile

```diff
@@ -1,61 +1,85 @@
-#### Common Build Image Stage ####
+# To use or update to a ruby version, change {BASE_RUBY_IMAGE}
+ARG BASE_RUBY_IMAGE=ruby:2.7.1-alpine
+# BASE_PACKAGES_IMAGE will default to apply-for-teacher-training-gem-node-dependencies
+# building all layers above it if a value is not specidied during the build
+ARG BASE_PACKAGES_IMAGE=apply-for-teacher-training-gem-node-dependencies
 
-FROM ruby:2.7.1-alpine AS common-build
+# Stage 1: gem-node-packages, download gem and node packages.
+FROM ${BASE_RUBY_IMAGE} AS gem-node-packages
```

Comment:
I think the language around dependencies and packages here could do with one more revision.

We're building a thing in that's going to be the `BASE_PACKAGES_IMAGE` which is really `gem-node-dependencies` which is a slimmed-down version of `gem-node-packages`. 

- `gem` and `node` aren't equivalent — `gem` is a package but `node` is a runtime.
- we're mixing up `dependencies` and `packages`

I think we should call them `dependencies` throughout as `packages` implies OS packages.

Propose something like:

```
ARG RUBY_BASE_IMAGE_WITH_BUNDLED_GEMS_AND_NODE_MODULES=gems-and-node-modules-without-build-dependencies
...
from ${BASE_RUBY_IMAGE} gems-and-node-modules
...
from ${BASE_RUBY_IMAGE} gems-and-node-modules-without-build-dependencies
...
from ${BASE_RUBY_IMAGE_WITH_BUNDLED_GEMS_AND_NODE_MODULES} assets-precompile
...
from ${BASE_RUBY_IMAGE} production
```

Reply:
thanks @duncanjbrown have renamed as suggested, but have kept the second layer name as `apply-for-teacher-training-gems-node-modules` instead of `gems-and-node-modules-without-build-dependencies` to keep is same as the name of the image being published.